minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -0,0 +1,331 @@
+            writer.println("nifi.content.claim.max.flow.files=" + getValueString(contentRepo, CONTENT_CLAIM_MAX_FLOW_FILES_KEY));
+            writer.println("nifi.content.repository.directory.default=./content_repository");
+            writer.println("nifi.content.repository.always.sync=" + getValueString(contentRepo, ALWAYS_SYNC_KEY));
+            writer.println();
+            writer.println("# Component Status Repository");
+            writer.println("nifi.components.status.repository.implementation=org.apache.nifi.controller.status.history.VolatileComponentStatusRepository");
+            writer.println("nifi.components.status.repository.buffer.size=" + getValueString(componentStatusRepo, BUFFER_SIZE_KEY));
+            writer.println("nifi.components.status.snapshot.frequency=" + getValueString(componentStatusRepo, SNAPSHOT_FREQUENCY_KEY));
+            writer.println();
+            writer.println("# web properties #");
+            writer.println("nifi.web.war.directory=./lib");
+            writer.println("nifi.web.http.host=");
+            writer.println("nifi.web.http.port=8081");
+            writer.println("nifi.web.https.host=");
+            writer.println("nifi.web.https.port=");
+            writer.println("nifi.web.jetty.working.directory=./work/jetty");
+            writer.println("nifi.web.jetty.threads=200");
+            writer.println();
+            writer.println("# security properties #");
+            writer.println("nifi.sensitive.props.key=" + getValueString(sensitiveProperties, SENSITIVE_PROPS_KEY__KEY));
+            writer.println("nifi.sensitive.props.algorithm=" + getValueString(sensitiveProperties, SENSITIVE_PROPS_ALGORITHM_KEY));
+            writer.println("nifi.sensitive.props.provider=" + getValueString(sensitiveProperties, SENSITIVE_PROPS_PROVIDER_KEY));
+            writer.println();
+            writer.println("nifi.security.keystore=" + getValueString(securityProperties, KEYSTORE_KEY));
+            writer.println("nifi.security.keystoreType=" + getValueString(securityProperties, KEYSTORE_TYPE_KEY));
+            writer.println("nifi.security.keystorePasswd=" + getValueString(securityProperties, KEYSTORE_PASSWORD_KEY));
+            writer.println("nifi.security.keyPasswd=" + getValueString(securityProperties, KEY_PASSWORD_KEY));
+            writer.println("nifi.security.truststore=" + getValueString(securityProperties, TRUSTSTORE_KEY));
+            writer.println("nifi.security.truststoreType=" + getValueString(securityProperties, TRUSTSTORE_TYPE_KEY));
+            writer.println("nifi.security.truststorePasswd=" + getValueString(securityProperties, TRUSTSTORE_PASSWORD_KEY));
+        } finally {
+            if (writer != null){
+                writer.flush();
+                writer.close();
+            }
+        }
+    }
+
+    private static void writeFlowXml(Map<String, Object> topLevelYaml, String path) throws IOException {
```

Comment:
This should probably make use of the built in tools for marshalling/unmarshalling XML that NiFi uses.  With things like the reporting task for provenance and possibly more complex flows, it would give a good bit more flexibility moving forward.